code/modules/mob/living/carbon/carbon.dm

```diff
@@ -1348,18 +1348,30 @@
 /mob/living/carbon/proc/check_signables_state()
 	var/obj/item/bodypart/left_arm = get_bodypart(BODY_ZONE_L_ARM)
 	var/obj/item/bodypart/right_arm = get_bodypart(BODY_ZONE_R_ARM)
-	var/empty_indexes = get_empty_held_indexes()
-	var/exit_right = (!right_arm || right_arm.bodypart_disabled)
-	var/exit_left = (!left_arm || left_arm.bodypart_disabled)
-	if(length(empty_indexes) == 0 || (length(empty_indexes) < 2 && (exit_left || exit_right)))//All existing hands full, can't sign
-		return SIGN_HANDS_FULL // These aren't booleans
-	if(exit_left && exit_right)//Can't sign with no arms!
-		return SIGN_ARMLESS
+	var/available_hands = length(get_empty_held_indexes()) // missing or disabled arms will count as empty
+	var/no_right_hand = (!right_arm || right_arm.bodypart_disabled)
+	var/no_left_hand = (!left_arm || left_arm.bodypart_disabled)
+
+	// subtract these from our available_hands
+	if(no_right_hand)
+		available_hands -= 1
+	if(no_left_hand)
+		available_hands -= 1
```

Comment:
I don't know how well our support for more than 2 hands is anymore, but I thought it might still be a good idea to try to maintain the possibility of it being a thing in the future.

```suggestion
	var/list/hands = get_empty_held_indexes()
	var/available_hands = length(hands)
	for(var/obj/item/bodypart/index as anything in hands)
		if(!index || index.bodypart_disabled)
			available_hands--
```

Reply:
Some caveats apply here.  `get_empty_held_indexes()` does the following:

https://github.com/tgstation/tgstation/blob/c1d5564f9bad83025c192c55a13b1639e7c53a69/code/modules/mob/inventory.dm#L75-L80

The list that is returned is just a array containing indexes (not `/obj/item/bodypart`) that don't contain an item.  Our mob would look something like this:

```
// this mob isn't holding any items
mob.held_items[1] = null
mob.held_items[2] = null
length(mob.get_empty_held_indexes()) // would return 2

// this mob is holding one item
mob.held_items[1] = /obj/item/pen
mob.held_items[2] = null
length(mob.get_empty_held_indexes()) // would return 1

// this mob is holding two item
mob.held_items[1] = /obj/item/pen
mob.held_items[2] = /obj/item/paper
length(mob.get_empty_held_indexes()) // would return 0
```

`get_empty_held_indexes` also doesn't take into account if an arm is disabled or cutoff and will still count as an empty item.  (so if both hands are cutoff it will return `2` empty indexes)

```
	var/list/hands = get_empty_held_indexes() // this list is just list(1, 2) not list(obj/item/bodypart, obj/item/bodypart)
	var/available_hands = length(hands)
	for(var/obj/item/bodypart/index as anything in hands)
               // this won't work since the hands list does not contain the bodyparts for the mob
		if(!index || index.bodypart_disabled)
			available_hands--
```

Reply:
On further code diving I might be able to use this code to get that effect:

https://github.com/tgstation/tgstation/blob/c1d5564f9bad83025c192c55a13b1639e7c53a69/code/modules/mob/inventory.dm#L471-L490

I'll test it later today when I get home.